ddtrace/contrib/internal/trace_utils.py

```diff
@@ -172,6 +172,30 @@ def _get_request_header_user_agent(headers, headers_are_case_sensitive=False):
     return ""
 
 
+def _get_request_header_referrer_host(headers, headers_are_case_sensitive=False):
+    # type: (Mapping[str, str], bool) -> str
+    """Get referrer host from request headers
+    :param headers: A dict of http headers to be stored in the span
+    :type headers: dict or list
+    :param headers_are_case_sensitive: Whether the headers are case sensitive
+    :type headers_are_case_sensitive: bool
+    :return: The referrer host if found, empty string otherwise
+    :rtype: str
+    """
+    if headers_are_case_sensitive:
+        referrer = _get_header_value_case_insensitive(headers, http.REFERER_HEADER)
+    else:
+        referrer = headers.get(http.REFERER_HEADER)
+    if referrer:
+        try:
+            parsed_url = parse.urlparse(referrer)
+            if parsed_url.hostname:
+                return parsed_url.hostname
+        except (ValueError, AttributeError):
+            log.debug("Failed to parse referer value: %s", referrer)
```

Comment:
[Non-blocking] I would avoid printing a message here, it can easily flood the debug log with something the user and us cannot "act" upon.